src/state_machine_tests.zig

```diff
@@ -2828,66 +2825,59 @@ test "get_change_events" {
     );
 }
 
-// Sanity test to check the maximum batch size on a 1MiB message.
+// Sanity test to check the maximum batch size.
 // For a comprehensive test of all operations, see the `input_valid` test.
 test "StateMachine: batch_elements_max" {
-    // TODO: Zig 0.13.0 lazy compilation allowed creating a state machine
-    // with `test_min` config and a 1MiB message. With Zig 0.14.1, many
-    // assertions are hit during compile time.
-    // We should either move this test to another non-testing binary or isolate
-    // the code in another type that doesn’t require creating the state machine.
-    if (true) return error.SkipZigTest;
```

Comment:
State machines specialized by configuration are overly complex and not fully functional. For example, overriding `message_body_size_max` did not work here.